src/wp-includes/class-wp-image-editor.php

```diff
@@ -367,14 +367,14 @@ protected function get_output_format( $filename = null, $mime_type = null ) {
 		 *
 		 * @param string[] $output_format {
 		 *     An array of mime type mappings. Maps a source mime type to a new
-		 *     destination mime type. Default empty array.
+		 *     destination mime type. Default is array( 'image/jpeg' => 'image/webp' ).
 		 *
 		 *     @type string ...$0 The new mime type.
 		 * }
 		 * @param string $filename  Path to the image.
 		 * @param string $mime_type The source image mime type.
 		 */
-		$output_format = apply_filters( 'image_editor_output_format', array(), $filename, $mime_type );
+		$output_format = apply_filters( 'image_editor_output_format', array( 'image/jpeg' => 'image/webp' ), $filename, $mime_type );
```

Comment:
I would suggest we don't change this here and instead add a callback function that we hook into this filter, to decouple the custom WebP transformation logic from the infrastructure here.

We could call this function e.g. `wp_image_editor_output_format_webp()` and put it in `wp-admin/includes/image.php`, then hook it into the filter within `wp-admin/includes/admin-filters.php`. For now it can just do what you have here, but it will allow us to easily customize further in a follow up PR, per size (in addition to passing the size name to the filter here). I reviewed the `WP_Image_Editor` code further and have an idea how to do the latter in a backward compatible way.

Overall, I'm in favor of decoupling in a filter callback, as that would also provide a more trivial way to unhook the whole logic, should a developer prefer to do so. If we add this before the filter executes we would then still wastefully execute logic.

Reply:
will update as a hook on the filter, good suggestion.

Reply:
Wound up naming the callback `wp_default_image_output_mapping` to decouple it from webp specifically, let me know what you think

Reply:
Naming SGTM, makes sense to generalize it.